packages/webapp-server/src/controllers/authController.ts

```diff
@@ -0,0 +1,173 @@
+      status: 403,
+    });
+  }
+
+  try {
+    console.log(`Searching for ${email} in the db`);
+    const user = await UserModel.findOne({ email });
+
+    console.log(`User ${user?.id} found.`);
+
+    if (!user) {
+      return next({
+        log: 'Attempted sign in failed, no matching user in db',
+        message: 'Invalid credentials',
+        status: 403,
+      });
+    }
+
+    console.log('Comparing password');
+    const userAuthenticated = await user.comparePassword(password);
+    if (!userAuthenticated) {
+      return next({
+        log: 'Attempted sign in failed, wrong password',
+        message: 'Invalid credentials',
+        status: 403,
+      });
+    }
+
+    console.log('Assigning req.user');
+
+    req.user = user;
+
+    return next();
+  } catch (error) {
+    return next({
+      log: 'Error occurred while attempting to sign in',
+      message: 'Unexpected error occurred',
+      status: 500,
+      error,
+    });
+  }
+};
+
+/**
+ * This middleware function will register a new user with the provided email and password. It will hash the password before saving it to the database. It will then set the user's id on the session and set the user on res.locals.user. It will then call next() to allow the request to continue.
+ * @param req
+ * @param res
+ * @param next
+ * @returns
+ */
+export const registerWithEmailAndPw = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  console.log('Registering new user...');
+
+  const { email, password } = req.body;
+  if (!email || !password) {
+    return next({
+      log: 'Email or password missing from request body.',
+      message: 'Invalid credentials',
+      status: 403,
+    });
+  }
+
+  // TODO: Validate email is in correct format and password is long enough with required characters
+
+  // TODO: Check if email is already in use
+
+  try {
+    console.log(`Creating user ${email}`);
+
+    const user = UserModel.build({
+      email,
+      password,
+      provider: 'local',
+    });
+
+    await user.save();
+
+    console.log('User created and saved');
+
+    req.user = user;
+
+    return next();
+  } catch (error) {
+    return next({
+      error,
+      log: 'An error occurred trying to create new user',
+      message: 'User account creation failed',
+      status: 500,
+    });
+  }
+};
+
+export const generateJwt = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const user = req.user as UserDocument;
+
+  if (!user) {
+    return next({
+      log: 'No user on request object, cannot create JWT',
+      message: 'You must log in to use this',
+      status: 403,
+    });
+  }
+
+  console.log(`Generating JWT for user ${user.email}`);
+
+  const token = jwt.sign(
+    {
+      sub: user.id,
+      email: user.email,
+    },
+    process.env.JWT_TOKEN || 'abc'
+  );
+
+  console.log(`Token generated: ${token}`);
+
+  res.cookie('podwatch_jwt', token, {
+    // httpOnly: true,
+    // secure: true,
```

Comment:
Can you see if these options can be reenabled?

Reply:
Yeah let me test that

Reply:
It does not let me access a protected route, the cookies object is blank. Let's retest if this is possible to include (httponly and secure) once we have frontend.